pkg/render/monitor.go

```diff
@@ -0,0 +1,270 @@
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+)
+
+const (
+	MonitoringAPIVersion = "monitoring.coreos.com/v1"
+
+	CalicoNodeAlertmanager = "calico-node-alertmanager"
+	CalicoNodeMonitor      = "calico-node-monitor"
+	CalicoNodePrometheus   = "calico-node-prometheus"
+	ElasticsearchMetrics   = "elasticearch-metrics"
+	FluentdMetrics         = "fluentd-metrics"
+	TigeraPrometheusDPRate = "tigera-prometheus-dp-rate"
+)
+
+func Monitor(
+	installation *operatorv1.InstallationSpec,
+	pullSecrets []*corev1.Secret,
+) Component {
+	return &monitorComponent{
+		installation: installation,
+		pullSecrets:  pullSecrets,
+	}
+}
+
+type monitorComponent struct {
+	installation      *operatorv1.InstallationSpec
+	pullSecrets       []*corev1.Secret
+	alertmanagerImage string
+	prometheusImage   string
+}
+
+func (mc *monitorComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := mc.installation.Registry
+	path := mc.installation.ImagePath
+	prefix := mc.installation.ImagePrefix
+
+	errMsgs := []string{}
+	var err error
+
+	mc.alertmanagerImage, err = components.GetReference(components.ComponentPrometheusAlertmanager, reg, path, prefix, is)
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+
+	mc.prometheusImage, err = components.GetReference(components.ComponentPrometheus, reg, path, prefix, is)
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+	return nil
+}
+
+func (mc *monitorComponent) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (mc *monitorComponent) Objects() ([]client.Object, []client.Object) {
+	objs := []client.Object{
+		createNamespace(common.TigeraPrometheusNamespace, mc.installation.KubernetesProvider),
+	}
+	objs = append(objs, secret.ToRuntimeObjects(secret.CopyToNamespace(common.TigeraPrometheusNamespace, mc.pullSecrets...)...)...)
+
+	objs = append(objs,
+		mc.alertmanager(),
+		mc.prometheus(),
+		mc.prometheusRule(),
+		mc.serviceMonitorCalicoNode(),
+		mc.serviceMonitorElasicsearch(),
+		mc.podMonitor(),
+	)
+
+	return objs, nil
+}
+
+func (mc *monitorComponent) Ready() bool {
+	return true
+}
+
+func (mc *monitorComponent) alertmanager() *monitoringv1.Alertmanager {
+	return &monitoringv1.Alertmanager{
+		TypeMeta: metav1.TypeMeta{Kind: monitoringv1.AlertmanagersKind, APIVersion: MonitoringAPIVersion},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      CalicoNodeAlertmanager,
+			Namespace: common.TigeraPrometheusNamespace,
+		},
+		Spec: monitoringv1.AlertmanagerSpec{
+			Image:        &mc.alertmanagerImage,
+			NodeSelector: map[string]string{"kubernetes.io/os": "linux"},
```

Comment:
Instead of adding this here I think you should be adding to the `ensureOSSchedulingRestrictions` function in `pkg/controller/utils/component.go`